jobs/integration/conftest.py

```diff
@@ -141,22 +153,58 @@ async def run(self, cmd, *args, stdin=None):
         if hasattr(stdin, "encode"):
             stdin = stdin.encode("utf8")
 
-        stdout, stderr = await proc.communicate(input=stdin)
-        if proc.returncode != 0:
+        stdout, stderr = bytearray(), bytearray()
+
+        def tee(line: bytes, sink: bytearray, fd: int):
+            sink += line
+            write = _tee == "out" and fd == 1
+            write |= _tee == "err" and fd == 2
+            if write or _tee is True:
+                os.write(fd, line)
+
+        async def _read_stream(stream, callback):
+            while True:
+                line = await stream.read()
```

Comment:
This blocks until EOF is received. So the output doesn't get tee'd until the subprocess closes stdout/stderr at its end. I'm assuming we want to see tee'd output before then.

See: [StreamReader.read](https://docs.python.org/3/library/asyncio-stream.html#asyncio.StreamReader.read). I think we need to pass a number to prevent it from waiting for EOF. In my testing with `read(1024)`, it blocks until at least 1 byte is received and then returns however many bytes are available, up to 1024. I think that should work for our purposes.

```suggestion
                line = await stream.read(1024)
```